src/parsers/texttracks/webvtt/html/create_settings_attributes.ts

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * How far from edge to start translating the text's origin,
+ * in order to keep all text inside it's box.
+ */
+import objectValues from "../../../../utils/object_values";
+
+const DEFAULT_HORIZONTAL_OFFSET = 30;
+const DEFAULT_VERTICAL_OFFSET = 5;
+
+export default function createSettingsAttributes (
+  settings : Partial<Record<string, string>>
+) : Attr {
+  const pAttr = document.createAttribute("style");
+  pAttr.value = getAttrValue(settings);
+  return pAttr;
+}
+
+const getAttrValue = (settings: Partial<Record<string, string>>) => {
+  const hasSettings = settings !== undefined && objectValues(settings).length !== 0;
+  if (!hasSettings) {
+    return "text-align:center";
+  }
+
+  return (
+    "position: absolute;" +
+    "margin: 0;" +
+    `${getTransformStyle(settings)}` +
+    `${getWidthStyle(settings.size)}` +
+    `${getPositionStyle(settings.position)}` +
+    `${getLineStyle(settings.line)}` +
+    `${getAlignStyle(settings.align)}`
+  );
+};
+
+const getTransformStyle = (settings: Partial<Record<string, string>>) => {
+  const xTranslateDefault = 50;
+  const yTranslateDefault = 50;
+
+  let xPosition = getPercentageValue(settings.position);
+  xPosition = xPosition !== null ?
+    xPosition :
+    xTranslateDefault;
+
+  let yPosition = getPercentageValue(settings.line);
+  yPosition = yPosition !== null ?
+    yPosition :
+    yTranslateDefault;
+
+  const width = getPercentageValue(settings.size);
+  const xOffset = width !== null ?
+    width :
+    DEFAULT_HORIZONTAL_OFFSET;
+  const yOffset = DEFAULT_VERTICAL_OFFSET;
+
+  const isCloseToXEdge = xPosition < xOffset || xPosition > (100 - xOffset);
+  const isCloseToYEdge = yPosition < yOffset || yPosition > (100 - yOffset);
+
+  const xTranslate = isCloseToXEdge ? xPosition : xTranslateDefault;
+  const yTranslate = isCloseToYEdge ? yPosition : yTranslateDefault;
+
+  return `transform: translate(-${xTranslate}%,-${yTranslate}%);`;
+
+} ;
+
+const getWidthStyle = (size: string | undefined) => {
+  return size !== undefined ?
+    `width:${size};` :
+    "";
+};
+
+const getPositionStyle = (positionSetting: string | undefined) => {
+  const positionDefault = 50;
```

Comment:
I think a default value is needed, because if the size of the cue box is set to something other than `100%` the `text-align` will not be enough for placing it correctly.